cpp/src/arrow/compute/kernels/cast-test.cc

```diff
@@ -1264,5 +1173,103 @@ TEST_F(TestCast, EmptyCasts) {
   }
 }
 
+// ----------------------------------------------------------------------
+// Test casting from NullType
+
+template <typename TestType>
+class TestNullCast : public TestCast {};
+
+typedef ::testing::Types<NullType, UInt8Type, Int8Type, UInt16Type, Int16Type, Int32Type,
+                         UInt32Type, UInt64Type, Int64Type, FloatType, DoubleType,
+                         Date32Type, Date64Type, FixedSizeBinaryType, BinaryType>
+    TestTypes;
+
+TYPED_TEST_CASE(TestNullCast, TestTypes);
+
+TYPED_TEST(TestNullCast, FromNull) {
+  // Null casts to everything
+  const int length = 10;
+
+  // Hack to get a DataType including for parametric types
+  std::shared_ptr<DataType> out_type =
+      TestBase::MakeRandomArray<typename TypeTraits<TypeParam>::ArrayType>(0, 0)->type();
+
+  NullArray arr(length);
+
+  std::shared_ptr<Array> result;
+  ASSERT_OK(Cast(&this->ctx_, arr, out_type, {}, &result));
+  ASSERT_OK(ValidateArray(*result));
+
+  ASSERT_TRUE(result->type()->Equals(*out_type));
+  ASSERT_EQ(length, result->length());
+  ASSERT_EQ(length, result->null_count());
+
+  // OK to look at bitmaps
+  ASSERT_ARRAYS_EQUAL(*result, *result);
```

Comment:
heh, why? Maybe extract the bitmap and ensure it's all zeroes?

Reply:
This is just copied from another test :-) I'm not sure how to do it simply otherwise...

Reply:
I mean, I'm not sure what this line does, might as well just remove it. I recall that the Equal implementation will early exit if pointers are equal.

Reply:
Ah, I see. Then I'll simply remove it. Hopefully validating the result array is enough.